### PR TITLE
Halfie Climb top left grapple teleports

### PR DESCRIPTION
**Pull request by blkerby**
This also includes a damage boost strat in HiJump E-Tank room, which I wanted to include before forgetting, even though it's not related.

The grapple teleports from this door lead to a lot of X-Ray Climb strats. Since it's so common, I made the earlier "Grapple Teleport X-Ray Climb" in Below Spazer (from Halfie Climb top right door) be no longer notable now.

### CODE REVIEW

**Review by osse101 — COMMENTED** [review]
Top position can get X=21 if you retract grapple about right as you touch the transition.

The X position being required for some tricks is real fuzzy.  Its mostly ok once you know about it, but most of the time it feels RNG.  

Breaking the camera after a grapple teleport is a problem and basically should be an additional tech imo.

File: region/maridia/outer/Crab Hole.json [hunk 1]
```diff
@@ -806,6 +806,28 @@
       },
       "requires": []
     },
+    {
+      "link": [3, 1],
+      "name": "Grapple Teleport X-Ray Climb",
```
[the remark this answers is not in the document]
Comment (blkerby) [reply region/maridia/outer/Crab Hole.json:811]: Ah, sure enough, it seems impossible; looks like I failed to test this one. At position 21, you get caught in the hole, while at position 19 or 20 you can't avoid the bottom transition. Removed these strats now.

**Comment by blkerby** [comment]
> Breaking the camera after a grapple teleport is a problem and basically should be an additional tech imo.

We have "canOffScreenMovement" for doing stuff off-camera? Or maybe you mean breaking graphics (permanently) when transitioning with a misaligned camera? I think it could make sense to add a tech "canBreakGraphics" for that; not that it takes any particular skill to break it, only to be able to tolerate the consequences afterward. It gets a little tricky because a lot of times it's preventable by doing the setup a little differently; we would have to go back and take another look at everything to see where there's a reliable way to prevent it and where there's not. Maybe it could be done in a follow-up later? Map Rando would probably ignore this tech anyway since it applies a patch to fix the graphics after every transition (except when using ultra-low QoL).

File: region/brinstar/green/Etecoon Energy Tank Room.json [hunk 2]
```diff
@@ -441,6 +461,65 @@
       "note": "Quick dropping through the crumbles allows reaching the door with more shinecharge time remaining.",
       "devNote": "Should there be a tech for crumble quick drop?"
     },
+    {
+      "link": [2, 3],
```
Comment (osse101) [inline region/brinstar/green/Etecoon Energy Tank Room.json:465]: [4, 3] versions of these as well

File: region/lowernorfair/east/The Worst Room In The Game.json [hunk 3]
```diff
@@ -742,6 +742,26 @@
...
+      "requires": [
+        "canXRayClimb",
+        {"heatFrames": 1600}
+      ],
+      "bypassesDoorShell": true,
+      "note": [
+        "Exit the previous room with Samus in a standing pose (while grappled).",
+        "After teleporting, press right to release Grapple while staying standing (not being forced into a crouch).",
+        "Then X-ray climb to get up to the door transition, without needing to open the door.",
+        "Climb quickly in order to avoid damage from the Pirate."
```
[the remark this answers is not in the document]
Comment (osse101) [reply region/lowernorfair/east/The Worst Room In The Game.json]: I wouldn't think there is RNG to it, but I don't know exactly how the transition to the "Look Around" state works.  I was trying to align my 5th climb with that timing.  It looks like I just wasn't going fast enough.

**Comment by osse101** [comment]
The offcamera xray climbs aren't a problem since you can turn off morph - in those cases offscreenmovement applies to when you need to count to an exact number.  And you can wiggle into transitions so its not like you need to commit to a spinjump check.

Permanently misaligning graphics would be about dealing with the consequences in future rooms, yeah.  You could have to do a tight hellrun with a handicap.  Maybe requiring a save to fix it could be an issue too, if there were no patch that fixes it.

Since camera-fix patch exists, it's not an urgent problem.

**Pull request merged**